fog/view/server/src/sharding_strategy.rs

```diff
@@ -48,6 +50,18 @@ impl EpochShardingStrategy {
     }
 }
 
+impl FromStr for EpochShardingStrategy {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Ok(block_range) = BlockRange::from_str(s) {
+            return Ok(Self::new(block_range));
+        }
+
+        Err("Invalid epoch sharding strategy.".to_string())
```

Comment:
Would it be better to propagate the block range parsing error?
```suggestion
        let block_range = BlockRange::from_str(s)?;
        Ok(Self::new(block_range))
```

Reply:
This suggestion only works when there's one sharding srategy. I'll leave it as is because this makes it easier to add a new strategy in the future (just add a new if block).